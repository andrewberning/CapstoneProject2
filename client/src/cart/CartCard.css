.CartCard {
  padding: 15px;
  margin-bottom: 15px;
}

.CartCard .item-link {
  display: flex;
  flex-direction: row;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.CartCard .item-img-container {
  width: 120px; 
  height: 120px; 
  margin-right: 20px;
  flex-shrink: 0;
}

.card-img {
  width: 100%;
  height: 100%;
  object-fit: cover; 
}

.card-body {
  flex-grow: 1;
  padding: 10px;
}

.cart-item-details .card-title {
  font-size: 1.25rem;
  font-weight: bold;
  margin-bottom: 10px;
}

.cart-item-details .card-price {
  font-size: 1.1rem;
  margin-bottom: 10px;
}

.cart-item-details .card-description {
  font-size: 0.9rem;
  color: #666;
  margin-bottom: 10px;
}

.card-quantity {
  font-size: 1rem;
  margin-bottom: 15px;
}

.item-quantity-container {
  display: flex;
  align-items: center;
  margin-top: 15px;
}

.item-quantity-container select {
  margin-right: 10px;
  padding: 5px;
  border-radius: 4px;
  border: 1px solid #ccc;
}

.item-quantity-container .btn-danger {
  padding: 5px 10px;
}

@media (max-width: 768px) {

}

@media (max-width: 480px) {

}