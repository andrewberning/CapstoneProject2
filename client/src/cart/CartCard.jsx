import { Link } from "react-router-dom";
import Card from "react-bootstrap/card"
// import Button from "react-bootstrap/Button";

export default function CartCard({ name, desc, price, img }) {

  return (
    <Link className="CartCard card d-flex flex-row mx-1 align-items-center" to={"/"}>
      <div className="cart-item-img">
        <img src={img} />
      </div>
      <Card.Body>
        <div className="cart-item-details">
        <Card.Title>{name}</Card.Title>
        <Card.Text>
          {price}
        </Card.Text>
        <Card.Text>
          {desc}
        </Card.Text>
        </div>
      </Card.Body>
    </Link>
  );
}