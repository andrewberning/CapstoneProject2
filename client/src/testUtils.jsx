import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
