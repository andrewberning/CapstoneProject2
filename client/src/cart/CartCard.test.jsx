import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CartCard from "./CartCard";
import { UserProvider } from "../testUtils";

describe('CategoriesBanner component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <CartCard />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <CartCard />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});