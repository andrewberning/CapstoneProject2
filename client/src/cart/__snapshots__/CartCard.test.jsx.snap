// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoriesBanner component > matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="CartCard card"
  >
    <a
      class="item-link"
      href="/undefined/undefined"
    >
      <div
        class="item-img-container"
      >
        <img
          class="card-img"
        />
      </div>
      <div
        class="card-body"
      >
        <div
          class="cart-item-details"
        >
          <div
            class="card-title"
          />
          <div
            class="card-price"
          >
            $
          </div>
          <div
            class="card-description"
          />
          <div
            class="card-quantity"
          >
            Qty: 
          </div>
        </div>
      </div>
    </a>
    <div
      class="item-quantity-container"
    >
      <select
        id="quantity-undefined"
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="5"
        >
          5
        </option>
        <option
          value="6"
        >
          6
        </option>
        <option
          value="7"
        >
          7
        </option>
        <option
          value="8"
        >
          8
        </option>
        <option
          value="9"
        >
          9
        </option>
        <option
          value="10"
        >
          10
        </option>
      </select>
    </div>
    <button
      class="btn btn-danger"
    >
      Remove
    </button>
  </li>
</DocumentFragment>
`;
